### PR TITLE
Adding test support for multiple platforms

### LINKED ISSUE #207: [FEATURE] Testing compatibility on different platforms
### Is your feature request related to a problem?
We test our clients for Linux platform but we do not know if these clients work smoothly on platforms like windows/mac etc.
Coming from https://github.com/opensearch-project/opensearch-clients/issues/18

### What solution would you like?
Test the client with most popular platforms and CPU architectures:

- x64
- ARM
- Linux
- Windows
- MacOS

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Adding test support across multiple platforms.

### Issues Resolved
Closes #207 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,10 +4,11 @@ on: [push, pull_request]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         java: [ 11 ]
+        os: [ubuntu-latest, windows-latest, macOS-latest]
```
Comment (saratvemulapalli) [inline .github/workflows/build.yml:11]: 🎉

Comment (saratvemulapalli) [reply .github/workflows/build.yml:11]: Do you know if this runs on MacOS M1 (arm64) or Intel (x64)?

**Review by saratvemulapalli — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
I think I merged this too fast. Looks like this doesn't run unit tests (it probably should) and those are in https://github.com/opensearch-project/opensearch-java/blob/main/.github/workflows/test-unit.yml.

**Comment by VachaShah** [comment]
> I think I merged this too fast. Looks like this doesn't run unit tests (it probably should) and those are in https://github.com/opensearch-project/opensearch-java/blob/main/.github/workflows/test-unit.yml.

Ah! I will add it in a new PR.

**Pull request merged**